### ISSUE TITLE
The main menu can be focused through keyboard navigation when hidden

### ISSUE THREAD

**Issue by tyrylu** [home-assistant/frontend#25371]
### Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.
- [x] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

It is still possible to reach the main menu and maybe some other components even when the sidebar is hidden. In that state a list is read by a screen reader, but not the items. The reading works properly when the sidebar is shown on screen.

### Describe the behavior you expected

The sidebar should be unreachable using keyboard navigation when hidden.

### Steps to reproduce the issue

1. Hide the sidebar
2. Try moving with the tab key around with a screen reader on (tried with Orca, but very likely affects other platforms as well)
3. Observe that you can reach the sidebar list.

### What version of Home Assistant Core has the issue?

2025.5

### In which browser are you experiencing the issue?

Firefox 138, latest Chromium as well.

### Which operating system are you using to run this browser?

Linux, but very likely does not matter

### State of relevant entities

```txt

```

### Problem-relevant frontend configuration

```yaml

```

### Javascript errors shown in your browser console/inspector

```txt

```
